src/pudl_archiver/cli.py

```diff
@@ -0,0 +1,140 @@
+
+logger = logging.getLogger("catalystcoop.pudl_archiver")
+
+
+def parse_main():
+    """Process base commands from the CLI."""
+    parser = argparse.ArgumentParser(description="Upload PUDL data archives to Zenodo")
+    parser.add_argument(
+        "datasets",
+        nargs="*",
+        help="Name of the Zenodo deposition. Supported: censusdp1tract, "
+        "eia860, eia861, eia923, eia_bulk_elec, epacems, epacamd_eia, "
+        "ferc1, ferc2, ferc6, ferc60, ferc714, eia860m",
+    )
+    parser.add_argument(
+        "--sandbox",
+        action="store_true",
+        help="Use Zenodo sandbox server",
+    )
+    parser.add_argument(
+        "--initialize",
+        action="store_true",
+        help="Initialize new deposition by preserving a DOI",
+    )
+    return parser.parse_args()
+
+
+async def archive_dataset(
+    name: str,
+    zenodo_client: ZenodoClient,
+    session: aiohttp.ClientSession,
+    initialize: bool = False,
+):
+    """Download and archive dataset on zenodo."""
+    async with zenodo_client.deposition_interface(name, initialize) as deposition:
+        # Create new deposition then return
+        match name:
+            case "ferc1":
+                archiver = Ferc1Archiver(session, deposition)
+            case "ferc2":
+                archiver = Ferc2Archiver(session, deposition)
+            case "ferc6":
+                archiver = Ferc6Archiver(session, deposition)
+            case "ferc60":
+                archiver = Ferc60Archiver(session, deposition)
+            case "ferc714":
+                archiver = Ferc714Archiver(session, deposition)
+            case "censusdp1tract":
+                archiver = CensusDp1TractArchiver(session, deposition)
+            case "eia860":
+                archiver = Eia860Archiver(session, deposition)
+            case "eia860m":
+                archiver = Eia860MArchiver(session, deposition)
+            case "eia861":
+                archiver = Eia861Archiver(session, deposition)
+            case "eia923":
+                archiver = Eia923Archiver(session, deposition)
+            case "eia_bulk_elec":
+                archiver = EiaBulkElecArchiver(session, deposition)
+            case "epacems":
+                archiver = EpaCemsArchiver(session, deposition)
+            case "epacmd_eia":
+                archiver = EpaCamdEiaArchiver(session, deposition)
+            case _:
+                raise RuntimeError("Dataset not supported")
+
+        await archiver.create_archive()
+
+
+async def archive_datasets():
+    """A CLI for the PUDL Zenodo Storage system."""
+    args = parse_main()
+    load_dotenv()
+
+    if args.sandbox:
+        upload_key = os.environ["ZENODO_SANDBOX_TOKEN_UPLOAD"]
+        publish_key = os.environ["ZENODO_SANDBOX_TOKEN_UPLOAD"]
+    else:
+        upload_key = os.environ["ZENODO_SANDBOX_TOKEN_UPLOAD"]
+        publish_key = os.environ["ZENODO_SANDBOX_TOKEN_UPLOAD"]
```

Comment:
Are these vars all supposed to be the same?

Reply:
No good catch, fixed now